cpp/src/parquet/arrow/writer.cc

```diff
@@ -319,6 +319,7 @@ class FileWriterImpl : public FileWriter {
       closed_ = true;
       if (row_group_writer_ != nullptr) {
         PARQUET_CATCH_NOT_OK(row_group_writer_->Close());
+        row_group_writer_ = nullptr;  // To avoid double closing
```

Comment:
Suggest:
```suggestion
      if (row_group_writer_ != nullptr) {
        auto row_group_writer = std::move(row_group_writer_);
        PARQUET_CATCH_NOT_OK(row_group_writer->Close());
```

Reply:
Also, a bit curious that when would it double-close here?

Reply:
> Also, a bit curious that when would it double-close here?

It will be closed again, in:
https://github.com/apache/arrow/blob/450175b2dbfc7834be72883c5a0e6ac93e8ed5aa/cpp/src/parquet/arrow/writer.cc#L308-L314

This is surely not intended normal use case (use after closed). Let me do one more check for why calling ```row_group_writer_->Close()``` twice also causes segfault and report back.

Reply:
You can leave this patch just checking here, and find out why it would close twice, then open a new issue for that?

I'm a bit tired today, maybe I'll gothrough the code here tomorrow

Reply:
You are right, we'd better open new issue for that.